docs/openstack_dataplanerole.md

```diff
@@ -78,6 +79,21 @@ NodeSection is a specification of the node attributes
 | ansibleVars | AnsibleVars for configuring ansible | string | false |
 | ansibleSSHPrivateKeySecret | AnsibleSSHPrivateKeySecret Private SSH Key secret containing private SSH key for connecting to node. Must be of the form: Secret.data.ssh-privatekey: <base64 encoded private key contents> https://kubernetes.io/docs/concepts/configuration/secret/#ssh-authentication-secrets | string | true |
 | extraMounts | ExtraMounts containing files which can be mounted into an Ansible Execution Pod | []storage.VolMounts | true |
+| nova | NovaTemplate specifies the parameters for the compute service deployment on the EDPM node. If this field is not specified then compute service will not be deployed on the EDPM node. | *[NovaTemplate](#novatemplate) | true |
+
+[Back to Custom Resources](#custom-resources)
+
+#### NovaTemplate
+
+NovaTemplate specifies the parameters for the compute service deployment on the EDPM node.
+
+| Field | Description | Scheme | Required |
+| ----- | ----------- | ------ | -------- |
+| cellName | CellName is the name of nova cell the compute(s) should be connected to | string | true |
+| novaInstance | NovaInstance is the name of the Nova CR that represents the deployment this compute belongs to | string | true |
+| customServiceConfig | CustomServiceConfig - customize the nova-compute service config using this parameter to change service defaults, or overwrite rendered information using raw OpenStack config format. The content gets added to to /etc/nova/nova.conf.d directory as 02-nova-override.conf file. | string | true |
+| defaultConfigOverwrite | DefaultConfigOverwrite - interface to overwrite default config files like e.g. logging.conf But can also be used to add additional files. Those get added to the service config dir in /etc/nova/ . | map[string]string | true |
+| deploy | Deploy true means the compute service(s) are allowed to be changed on the EDPM node if necessary. If set to false then only the pre-requisite data (e.g. config maps) will be generated but no actual modification on the compute node itself will happen. | bool | true |
```

Comment:
on the EDPM nodes of this role?

Reply:
NovaTemplate is use both in the Node and the Role so I need to word this carefully. I changed this to be more generic.

Reply:
What would the case be for exposing this deploy flag in the dataplane CRD's? We have deployStrategy.deploy on the dataplane CRD's already, can we just set NovaTemplate based on that value?

If there is a use case where we do need these set independently, then we need to account for the handling for when the values are not both true or both false, etc, given that all of the deployment logic is guarded by deployStrategy.deploy.

Reply:
adoption
we need to be able to have the dataplane operator create the NovaExternalComptue CR with deploy=false so that the generated config can be inspected without the ansibel being executed on the compute nodes.

if i understand correctly the current logic in the dataplane operator will not allow that to be created unless the deployStrategy.deploy=ture

Reply:
Yepp. If we need to trigger the config generation of the nova-compute service on the EDPM node without deploying that config (i.e. inspection during adoption) then we need to let the dataplane-operator create NovaExternalCompute CRs with deploy=false.

I'm not sure how the config generation, and config inspection, during adoption will happen to the rest of the EDPM services though. So if there be a separate mechanism for that then we can consider adopting to it.

Reply:
the dataplen operator has not been adapted to generating all configs using go templates yet.

I'm not sure when that is planned for but as a result there currently isn't a good way to see what configs it will generate as that is unfortunately still done by Ansible.
ideally it would be adapted to also generate the configs via go templates and just copy them but its a lot of work and I'm not sure we have people to work on it.